Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```diff
@@ -0,0 +1,414 @@
+                if let terminals = element.assetType.terminalConfiguration?.terminals {
+                    if terminals.count > 1 {
+                        terminalSelectorIsOpen.toggle()
+                        return
+                    } else {
+                        if let terminal = terminals.first {
+                            statusText = "Juntion element with terminal \(terminal.name) added to the filter barriers."
+                        }
+                    }
+                }
+            case .edge:
+                if let line = GeometryEngine.makeGeometry(from: geometry, z: nil) as? Polyline {
+                    element.fractionAlongEdge = GeometryEngine.polyline(
+                        line,
+                        fractionalLengthClosestTo: location,
+                        tolerance: -1
+                    )
+                    statusText = String(format: "Edge element at fractionAlongEdge %.3f added to the filter barriers.", element.fractionAlongEdge)
+                }
+            @unknown default:
+                return
+            }
+            
+            traceParameters.addFilterBarrier(element)
+            lastAddedElement = element
+            let point = geometry as? Point ?? location
+            addGraphic(for: point, traceLocationType: RunValveIsolationTraceView.Model.filterBarrierIdentifier)
+            resetEnabled = true
+            traceEnabled = true
+            hasFilterBarriers = true
+        }
+        
+        /// Adds the filter barrier of the user selected terminal to the trace parameters.
+        func addTerminal(to point: Point) {
+            if let terminal = lastAddedElement?.terminal, let lastAddedElement {
+                traceParameters.addFilterBarrier(lastAddedElement)
+                statusText = "Juntion element with terminal \(terminal.name) added to the filter barriers."
+                hasFilterBarriers = true
+                addGraphic(
+                    for: point,
+                    traceLocationType: RunValveIsolationTraceView.Model.filterBarrierIdentifier
+                )
+            }
+        }
+    }
+}
+
+private extension ArcGISCredential {
+    /// The public credentials for the data in this sample.
+    /// - Note: Never hardcode login information in a production application. This is done solely
+    /// for the sake of the sample.
+    static var publicSample: ArcGISCredential {
+        get async throws {
+            try await TokenCredential.credential(
+                for: .featureServiceURL,
+                username: "viewer01",
+                password: "I68VGU^nMurF"
+            )
+        }
+    }
+}
+
+extension UtilityCategory: Equatable {
+    public static func == (lhs: UtilityCategory, rhs: UtilityCategory) -> Bool {
+        lhs.name == rhs.name
+    }
+}
+
+extension UtilityCategory: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(name)
+    }
+}
```

Comment:
The samples app shouldn't be adding conformances like this to ArcGIS types. Instead, other means should be used, such as specifying the name as the id (`id: \.name`) and using identity comparison (`===`).

Reply:
I specified the name as the id and used identity comparison, but get an error when removing the hashable and equatable conformances. 
<img width="1119" alt="Screenshot 2023-06-27 at 9 35 41 AM" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/c05ba3f7-3f3b-45b1-968b-aa32c70a57c4">